server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Number.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import com.google.common.base.Preconditions;
+
+public class Number {
+    private static final int MIN_VALUE = 0;
+    private static final double MAX_INT_VALUE = Math.pow(2, 31);
+    private static final double MAX_LONG_VALUE = Math.pow(2, 53);
+
+    public static Number factory() {
+        return new Number();
+    }
+
+    private long value;
```

Comment:
This value should be final.

fromInt and fromLong should be static and you can delete factory()